MailContainerTest/Providers/MailContainerStoreProvider.cs

```diff
@@ -0,0 +1,32 @@
+﻿using MailContainerTest.Data;
+using Microsoft.Extensions.Configuration;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace MailContainerTest.Providers
+{
+    public class MailContainerStoreProvider : IMailContainerStoreProvider
+    {
+        private string _dataStoreType;
+
+        public MailContainerStoreProvider(IConfiguration configuration)
+        {
+            _dataStoreType = configuration["DataStoreType"];
```

Comment:
Would haved liked to look at having DataStoreType as an Enum instead of a magic string